src/Reflection/ReflectionProperty.php

```diff
@@ -361,4 +372,120 @@ public function __clone()
     {
         throw Uncloneable::fromClass(__CLASS__);
     }
+
+    /**
+     * @param object|null $object
+     * @return mixed
+     * @throws PropertyNotAccessible
+     * @throws ClassDoesNotExist
+     * @throws NoObjectProvided
+     * @throws NotAnObject
+     * @throws ObjectNotInstanceOfClass
+     */
+    public function getValue($object = null)
+    {
+        $this->assertAccessibility();
```

Comment:
IMO, our own implementation should drop the `setAccessible()` bullcrap

Reply:
Done. `setAccessible()` implemented only in adapters.